src/components/dashboard/Dashboard.js

```diff
@@ -1,5 +1,5 @@
 // @flow
-import React, { useEffect, useState } from 'react'
+import React, { useEffect, useMemo, useState } from 'react'
```

Comment:
General recommendations:
1. Use some events / callbacks to determine when animation actually ends
2. Don't use inline callbacks
3. memoize callbacks via useCallback
4. Component's logic is too complex and should be split to separate custom hooks (for animation, feeds management, dialogs etc)

Reply:
1. We already tried to manage it in a more proper way. We tried to use something like you recommend but it doesn't work as expected as there are some issues with List re-rendering so we decided to keep that hack (setTimeout, delays) at the moment as it took a lot of time already.
2. Will refactor it
3. Will do it
4. 100% Agreed. I think that a separate task should be created for the Dashboard refactor as it will take valuable time.